aws-auth-cognito/src/main/java/com/amplifyframework/statemachine/codegen/data/SignedInData.kt

```diff
@@ -31,13 +31,12 @@ internal data class SignedInData(
     override fun equals(other: Any?): Boolean {
         return if (super.equals(other)) {
             true
-        } else if (other == null || javaClass != other.javaClass) {
+        } else if (other == null || javaClass != other.javaClass || other !is SignedInData) {
```

Comment:
Wouldn't `javaClass != other.javaClass` be enough to rule out `other` not being of type SignedInData?

Reply:
It is but it doesn't add the cast safety required for the else block "as".